Running-Mastodon/Migration-guide.md

```diff
@@ -0,0 +1,79 @@
+
+**Note:** this guide was written with Ubuntu Server 16.04 in mind; your mileage may vary for other setups.
+
+What needs to be migrated
+---
+
+At a high level, you'll need to copy over the following:
+
+- The `~/live/public/system` directory, which contains user-uploaded images and videos
+- The Postgres database (using [pg\_dump](https://www.postgresql.org/docs/9.1/static/backup-dump.html))
+- The `~/live/.env.production` file, which contains server config and secrets
+
+Less crucially, you'll proably also want to copy the following for convenience:
+
+- The nginx config (under `/etc/nginx/sites-available/default`)
+- The systemd config files (`/etc/systemd/system/mastodon-*.service`), which may contain your server tweaks and customizations
+- The pgbouncer configuration under `/etc/pgbouncer` (if you're using it)
+
+Preliminary steps
+----
+
+All of the following steps can be taken without any downtime for your existing server.
+
+First, follow the [Production Guide](https://github.com/tootsuite/documentation/blob/master/Running-Mastodon/Production-guide.md) to set up a brand-new Mastodon server. However, don't set up a new Postgres database using `db:setup`; we're going to copy it over instead.
+
+### Dump and load Postgres
+
+Create an empty Postgres database using the `template0` database (which is useful when restoring a Postgres dump, [as described in the pg\_dump documentation](https://www.postgresql.org/docs/9.1/static/backup-dump.html#BACKUP-DUMP-RESTORE)). You'll want to run this as the `mastodon` user on your new system:
+
+```bash
+createdb -T template0 mastodon_production
+```
+
+Next, create a Postgres dump on the old system (again, as the `mastodon` user):
+
+```bash
+pg_dump mastodon_production > dump.sql
+```
+
+Copy the `dump.sql` file over, using `scp` or `rsync`. Then on the new system, run:
+
+```bash
+psql mastodon_production < dump.sql
+```
+
+This will copy the Postgres database from the old system to the new one. You may see a warning about `no privileges could be revoked for "public"`, but [you can ignore it](https://confluence.atlassian.com/bamkb/errors-or-warnings-appear-when-importing-postgres-database-dump-829036698.html).
+
+Note that you will need to re-run this process from scratch (i.e. with a new database) if the old database changes after the dump. So you may want to use a test database first to play around with.
+
+### Copy `system/` files
+
+This will probably take a while, and you'll want to avoid re-copying unnecessarily, so using `rsync` is recommended. On your old machine, as the `mastodon` user:
+
+```bash
+rsync -avz ~/live/public/system/ mastodon@example.com:~/live/public/system/
+```
+
+You'll want to re-run this if any of the files on the old server change.
+
+You can also copy over any other important files, such as `.env.production` and the nginx, systemd, and pgbouncer config files.
+
+Migration process
```

Comment:
I would put this on top, so that the process is easily accessible, and then the full article gives some detail on each step. Apart from that it looks good to me. When I did my migration in july I did pretty much the same thing, except instead of making a brand-new install of mastodon I just moved the entire `live/` folder and migrated the DB and that was it. This method is probably more resilient tho.

Reply:
OK, I moved the simple steps to the top and removed some excess verbiage. Hopefully it's easier to follow now!